check_sites.py

```diff
@@ -18,33 +18,48 @@
 import os
 from publicsuffix2 import PublicSuffixList
 
+def find_diff_sets(old_sets, new_sets):
+    diff_sets = {}
+    subtracted_sets = {}
+    difference = set(old_sets)^set(new_sets)
+    for diff in difference:
+        if diff in new_sets:
+            diff_sets[diff] = new_sets[diff]
+        else:
+            subtracted_sets[diff] = old_sets[diff]
+    return diff_sets, subtracted_sets
+
 
 def main():
     args = sys.argv[1:]
-    inputFile = 'first_party_sets.JSON'
-    inputPrefix = ''
-    opts, _ = getopt.getopt(args, "i:", ["data_directory="])
+    input_file = 'first_party_sets.JSON'
+    input_prefix = ''
+    with_diff = False
+    opts, _ = getopt.getopt(args, "i:", ["data_directory=", "with_diff="])
     for opt, arg in opts:
         if opt == '-i':
-            inputFile = arg
+            input_file = arg
         if opt == '--data_directory':
-            inputPrefix = arg
+            input_prefix = arg
+        if opt == '--with_diff':
+            with_diff = arg.lower() == "true"
 
-    # Open the canonical sites, and load the json
-    with open(inputFile) as f:
+    # Open and load the json of the new list
+    with open(input_file) as f:
         try:
             fps_sites = json.load(f)
         except Exception as inst:
         # If the file cannot be loaded, we will not run any other checks
-            print("There was an error when loading "+ inputFile + 
+            print("There was an error when loading "+ input_file + 
                   "\nerror was: " + inst)
-            exit()      
+            exit()  
```

Comment:
```suggestion
            return  
```

Reply:
Happy to make this change, but is there a specific reason why one would be preferred over the other?

Reply:
I have a slight preference for making the control flow as "local" as possible, just to minimize "spooky action at a distance". But it's not a strong preference here, since this is in the `main` function after all.